spring-cloud-dataflow-docs/src/main/asciidoc/tasks.adoc

```diff
@@ -1159,4 +1160,72 @@ Spring Cloud Data Flow will have a suffix appended to it consisting of `-scdf-<t
+
+image::{dataflow-asciidoc}/images/dataflow-task-default-version.png[Task Application Default Version, scaledwidth="50%"]
+
+=== Task Launch Lifecycle
+In previous versions of Spring Cloud Data Flow, when the request to launch a task was received, Spring Cloud Data Flow would deploy the application (if needed) and execute it.  If the application was being run on a plaform that did not need to have the application deployed every time (CloudFoundry for example), the previously deployed application was used.  This flow has changed in 2.3.  Let's take a look at what happens when a task launch request comes in now.
+
+image::{dataflow-asciidoc}/images/dataflow-task-launch-flow.png[Flow For Launching A Task, scaledwidth="50%"]
+
+There are three main flows to consider in the above diagram.  Launching the first time/launching with no changes is one, the other is launching when there are changes.  We'll look at the flow with no changes first.
+
+==== Launch a Task With No Changes
+1. A launch request comes into to Data Flow.  Data Flow determines that an upgrade is not required since nothing has changed (no properties, deployment properites, or versions have changed since the last execution).
+
+[start=5]
+5. On platforms that cache a deployed artifact (CloudFoundry at the writing of this documentation), Data Flow checks if the application was previously deployed.
+6. If the application needs to be deployed, Data Flow will do the deployment of the task application.
+7. Data Flow launches the application.
+
+That flow is the default behavior and will occur every time a request comes in if nothing has changed.  It is important to note that this is the same flow that Data Flow has always executed for launching of tasks.
+
+==== Launch a Task With Changes That Is Not Currently Running
+The second flow to consider when launching a task is if there was a change in any of the task application version, application properties, or deployment properties.  In this case, the following flow is executed
+
+1. A launch request comes into to Data Flow.  Data Flow determines that an upgrade is required since there was a change in either task application version, application properties, or deployment properties.
+2. Data Flow checks to see if another instance of the task definition is currently running.
+
+[start=4]
+4. If there is not another instance of the task definition currently running, the old deployment is deleted.
+5. On platforms that cache a deployed artifact (CloudFoundry at the writing of this documentation), Data Flow checks if the application was previously deployed (this check will always be no in this flow).
+6. Data Flow will do the deployment of the task application with the updated values (new application version, new merged properties, new merged deployment properties).
```

Comment:
Change "will do" to "does". Add "and" before the last item in the list that is within parentheses.